router/batchrouter/batchrouter_test.go

```diff
@@ -517,3 +528,162 @@ func TestPostToWarehouse(t *testing.T) {
+		customTimezone string
+		filenamePrefix string
+	}{
+		{
+			name:           "default",
+			filenamePrefix: "rudder-logs/{sourceID}/2021-06-28",
+		},
+		{
+			name:           "CET",
+			customTimezone: "Europe/Amsterdam",
+			filenamePrefix: "rudder-logs/{sourceID}/2021-06-28",
+		},
+		{
+			name:           "IST",
+			customTimezone: "Asia/Kolkata",
+			filenamePrefix: "rudder-logs/{sourceID}/2021-06-29",
+		},
+	}
+
+	// time is picked so it goes to the next day for IST
+	now := time.Date(2021, 6, 28, 21, 1, 30, 0, time.UTC)
+
+	var jobs []*jobsdb.JobT
+	bcs := make(map[string]backendconfig.ConfigT)
+	filePrefixes := make([]string, 0)
+
+	for _, tc := range testCases {
+		workspaceID := `workspaceID` + tc.name
+
+		if tc.customTimezone != "" {
+			c.Set("BatchRouter.customTimezone."+workspaceID, tc.customTimezone)
+		}
+
+		s3Dest := destination.MINIOFromResource("minio-dest"+tc.name, minioResource)
+		s3Dest.WorkspaceID = workspaceID
+		bc := backendconfigtest.NewConfigBuilder().WithSource(
+			backendconfigtest.NewSourceBuilder().WithConnection(s3Dest).Build(),
+		).Build()
+		bc.WorkspaceID = workspaceID
+
+		bcs[workspaceID] = bc
+
+		filePrefixes = append(filePrefixes, strings.ReplaceAll(tc.filenamePrefix, "{sourceID}", bc.Sources[0].ID))
+
+		jobs = append(jobs, &jobsdb.JobT{
+			WorkspaceId: workspaceID,
+			EventPayload: jsonb.RawMessage(`
+			{
+				"receivedAt": "2019-10-12T07:20:50.52Z",
+				"metadata": {
+					"columns": {
+						"id": "string"
+					},
+					"table": "tracks"
+				}
+			}`),
+			Parameters: jsonb.RawMessage([]byte(fmt.Sprintf(`{
+				"source_id": %[1]q,
+				"destination_id": %[2]q,
+				"receivedAt": %[3]q
+			}`, bc.Sources[0].ID, s3Dest.ID, time.Now().Format(time.RFC3339)))),
+			CustomVal: s3Dest.DestinationDefinition.Name,
+			CreatedAt: time.Now(),
+		})
+	}
+
+	batchrouter := &Handle{
+		now: func() time.Time {
+			return now
+		},
+	}
+	batchrouter.Setup(
+		"MINIO",
+		backendconfigtest.NewStaticLibrary(bcs),
+		routerDB,
+		errDB,
+		nil,
+		transientsource.NewEmptyService(),
+		rsources.NewNoOpService(),
+		destinationdebugger.NewNoOpService(),
+		c,
+	)
+
+	batchrouter.minIdleSleep = config.SingleValueLoader(time.Microsecond)
+	batchrouter.uploadFreq = config.SingleValueLoader(time.Microsecond)
+	batchrouter.mainLoopFreq = config.SingleValueLoader(time.Microsecond)
+
+	err = routerDB.Store(context.Background(), jobs)
+	require.NoError(t, err)
+
+	batchrouter.Start()
+	defer batchrouter.Shutdown()
+
+	require.Eventually(t, func() bool {
+		return len(minioContents(t, context.Background(), minioResource, "")) == len(bcs)
+	}, 5*time.Second, 200*time.Millisecond)
+
+	filenames := lo.Map(
+		lo.Keys(minioContents(t, context.Background(), minioResource, "")),
+		func(k string, _ int) string { return path.Dir(k) },
+	)
+
+	require.ElementsMatch(t, filePrefixes, filenames)
+}
+
+func minioContents(t require.TestingT, ctx context.Context, dest *minio.Resource, prefix string) map[string]string {
```

Comment:
It seems to be similar to what we have in [archiver](https://github.com/rudderlabs/rudder-server/blob/f3532aee89d1e4e9ac9597fdee2a7a82bbc9ec2e/warehouse/archive/archiver_test.go#L214). Can we move this to test helper so that we can use it in both places.